### ISSUE TITLE
ModuleNotFoundError: No module named 'aiogithubapi'

### ISSUE THREAD

**Issue by edemiguel** [hacs/integration#818]
This is not an old issue as you can see in the following links:
[Setup fails after new installation: module aiogithubapi missing](https://github.com/hacs/integration/issues/421)
[No module named 'aiogithubapi'](https://github.com/hacs/integration/issues/662)
I'm trying to install HACS but the error is not leting me.
I've restarted home assistant a few times allready and it hasn't helped.
I don't know much about python but I tried installling aiogithubapi manually `python3 -m pip install -U aiogithubapi` but it didn't work.
For your information I'm running the homeassistant with it's own user (called homeassistant). I installed the home assistant following [Manual installation on a Raspberry Pi](https://www.home-assistant.io/docs/installation/raspberry-pi/) I don't know if this is related to this problem .

This is the home-assistant.log

```
2019-11-23 16:42:03 ERROR (MainThread) [homeassistant.loader] Error loading custom_components.websocket_api. Make sure all dependencies are installed
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/loader.py", line 355, in _load_file
    module = importlib.import_module(path)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/homeassistant/.homeassistant/custom_components/__init__.py", line 8, in <module>
    from aiogithubapi import AIOGitHub
ModuleNotFoundError: No module named 'aiogithubapi'
```

**Comment by edemiguel** [comment]
By the way, I'm using HACS 0.17.4 with Home assitant 0.102.1

**Comment by ludeeus** [comment]
This is not an issue with the integration.
Home Assistant is responsible to install requirements, if it does not you need to troubleshoot your Home Assistant installation.

**Comment by edemiguel** [comment]
Thank you ludeeus! I've reported it and opened a new issue at [ModuleNotFoundError: No module named 'aiogithubapi'](https://github.com/home-assistant/home-assistant/issues/29011).
While someone offer me a solution, I started to install in the virtual enviroment the modules needed. Until now I've installed some requested modules with the following commands:

```
source /srv/homeassistant/bin/activate
python3 -m pip install -U aiogithubapi
python3 -m pip install -U semantic_version
python3 -m pip install -U integrationhelper
```
Now I'm stuck installing with the custom_components.hacs module and the custom_components.hassio module

```
homeassistant@raspberrypi:/home/pi $ source /srv/homeassistant/bin/activate
(homeassistant) homeassistant@raspberrypi:/home/pi $ python3 -m pip install -U custom_components.hacs
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting custom_components.hacs
Could not install packages due to an EnvironmentError: 404 Client Error: Not Found for url: https://pypi.org/simple/custom-components-hacs/

(homeassistant) homeassistant@raspberrypi:/home/pi $ python3 -m pip install -U custom_components.hassio
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting custom_components.hassio
Could not install packages due to an EnvironmentError: 404 Client Error: Not Found for url: https://pypi.org/simple/custom-components-hassio/

```
And this is the home-assistant.log l'm getting

```
2019-11-24 01:17:55 ERROR (MainThread) [homeassistant.loader] Error loading custom_components.websocket_api. Make sure all dependencies are installed
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/loader.py", line 346, in _load_file
    module = importlib.import_module(path)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/homeassistant/.homeassistant/custom_components/__init__.py", line 20, in <module>
    from .hacsbase.configuration import Configuration
  File "/home/homeassistant/.homeassistant/custom_components/hacsbase/configuration.py", line 4, in <module>
    from custom_components.hacs.hacsbase.exceptions import HacsUserScrewupException
ModuleNotFoundError: No module named 'custom_components.hacs'
2019-11-24 12:56:02 ERROR (MainThread) [homeassistant.loader] Error loading custom_components.hassio. Make sure all dependencies are installed
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/loader.py", line 346, in _load_file
    module = importlib.import_module(path)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/homeassistant/.homeassistant/custom_components/__init__.py", line 20, in <module>
    from .hacsbase.configuration import Configuration
  File "/home/homeassistant/.homeassistant/custom_components/hacsbase/configuration.py", line 4, in <module>
    from custom_components.hacs.hacsbase.exceptions import HacsUserScrewupException
ModuleNotFoundError: No module named 'custom_components.hacs'
2019-11-24 12:56:05 WARNING (Thread-2) [pychromecast.socket_client] [Comedor:8009] Heartbeat timeout, resetting connection
```

**Comment by edemiguel** [comment]
In the previous log you can see that I get a 404 from the custom_components.hacs module and the custom_components.hassio module while It tries to download them.

**Comment by neuman1812** [comment]
Just found this after trying to install HACS,  Im on a hassbian install and running into the same issue.   I was able to install it under venv,  After a reboot,  HA still does not see this module.

Everything else on my HA works.  Except this.  Any suggestion at all wound be helpfull.

```

(homeassistant) pi@hassbian:/home/homeassistant/.homeassistant/www $ sudo python3 -m pip install -U aiogithubapi
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting aiogithubapi
  Downloading https://files.pythonhosted.org/packages/69/09/9a0be3211526c556f33be342657583cd0ca179ad97693082daf72b520c2b/aiogithubapi-0.4.2-py3-none-any.whl
Collecting aiohttp (from aiogithubapi)
  Downloading https://files.pythonhosted.org/packages/c2/f7/f0ad3dbace4762fef5d80aa4124b41bf218e4c4dd6d387a86cede707d9a4/aiohttp-3.6.2-py3-none-any.whl (441kB)
    100% |████████████████████████████████| 450kB 266kB/s 
Collecting backoff (from aiogithubapi)
  Downloading https://files.pythonhosted.org/packages/8a/87/e9b3cba3a27e14a90e913a13122bb235f7b0c1215b022d709bcde963b367/backoff-1.9.2-py2.py3-none-any.whl
Collecting async-timeout (from aiogithubapi)
  Downloading https://files.pythonhosted.org/packages/e1/1e/5a4441be21b0726c4464f3f23c8b19628372f606755a9d2e46c187e65ec4/async_timeout-3.0.1-py3-none-any.whl
Collecting attrs>=17.3.0 (from aiohttp->aiogithubapi)
  Downloading https://files.pythonhosted.org/packages/a2/db/4313ab3be961f7a763066401fb77f7748373b6094076ae2bda2806988af6/attrs-19.3.0-py2.py3-none-any.whl
Collecting chardet<4.0,>=2.0 (from aiohttp->aiogithubapi)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
    100% |████████████████████████████████| 143kB 723kB/s 
Collecting multidict<5.0,>=4.5 (from aiohttp->aiogithubapi)
  Downloading https://www.piwheels.org/simple/multidict/multidict-4.6.1-cp37-cp37m-linux_armv7l.whl (389kB)
    100% |████████████████████████████████| 399kB 271kB/s 
Collecting yarl<2.0,>=1.0 (from aiohttp->aiogithubapi)
  Downloading https://www.piwheels.org/simple/yarl/yarl-1.3.0-cp37-cp37m-linux_armv7l.whl (255kB)
    100% |████████████████████████████████| 256kB 378kB/s 
Requirement already satisfied, skipping upgrade: idna>=2.0 in /usr/lib/python3/dist-packages (from yarl<2.0,>=1.0->aiohttp->aiogithubapi) (2.6)
Installing collected packages: attrs, chardet, multidict, async-timeout, yarl, aiohttp, backoff, aiogithubapi
Successfully installed aiogithubapi-0.4.2 aiohttp-3.6.2 async-timeout-3.0.1 attrs-19.3.0 backoff-1.9.2 chardet-3.0.4 multidict-4.6.1 yarl-1.3.0

```

**Comment by neuman1812** [comment]
OK...my issue was resolved.  I figured it out. 

The hacs.zip file.  I downloaded to the custom_componets folder  and used a simple unzip command to open it.  It puts all the files in  custom_components/* folder.

HA...doesn't know how to handle that.

I manually created the hacs folder and unzipped it in there.  After a restart of HA,  everything installed and works correctly now.

The correct command on a Linux install would be:

unzip -d hacs hacs.zip
